### PR TITLE
Reinstate debugging tutorial

### PR DESCRIPTION
**Pull request by johnwhitington**
There is a decent debugging tutorial in `site/learn/tutorials/debug.md` which appears to be unused.

This PR includes a lightly edited version of it, and links it back into the tutorials.

### CODE REVIEW

File: site/learn/tutorials/debug.md [hunk 1]
```diff
@@ -3,16 +3,18 @@
...
+  which works in the interactive top level.
+* The [OCaml debugger](#The-OCaml-debugger), which allows analysing programs
   compiled with `ocamlc`.
 
-## Tracing functions calls in the toplevel
-The simplest way to debug programs in the toplevel is to follow the
-function calls, by “tracing” the faulty function:
+## Tracing functions calls in the top level
+
+The simplest way to debug programs in the top level is to follow the function
```
Comment (avsm) [inline site/learn/tutorials/debug.md]: A nit: the OCaml manual calls it a "toplevel" (https://ocaml.org/manual/toplevel.html) but you split it up to "top level" in this PR.

File: site/learn/tutorials/debug.md [hunk 2]
```diff
@@ -163,51 +168,56 @@ Program end.
...
 ```ocaml
 print_string (find_address "INRIA"); print_newline ();;
                                   ^
 ```
-To sum up: if you're developping a program you can compile it with the
-`-g` option, to be ready to debug the program if necessary. Hence, to
-find a spurious exception you just need to type `ocamldebug a.out`, then
-`r`, `b`, and `bt` gives you the backtrace.
+
+To sum up: if you're developing a program you can compile it with the `-g`
```
Comment (avsm) [inline site/learn/tutorials/debug.md:214]:
```suggestion
To sum up: if you're developing a program, you can compile it with the `-g`
```

**Review by avsm — APPROVED** [review]

File: site/learn/tutorials/debug.md (hunk 1 above)
Comment (johnwhitington) [reply site/learn/tutorials/debug.md]: Fixed. Thanks.

**Comment by avsm** [comment]
thanks!

**Pull request merged**